clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -42,7 +42,7 @@ class KubernetesServerGroupCachingAgent implements CachingAgent, OnDemandAgent,
   @Deprecated
   private static final String LEGACY_ON_DEMAND_TYPE = 'KubernetesCluster'
 
-  private static final String ON_DEMAND_TYPE = 'Cluster'
+  private static final List<String> ON_DEMAND_TYPES = ['ServerGroup', 'Cluster', 'Instance']
```

Comment:
I only did a quick read of this, but I would have thought this would be `ON_DEMAND_TYPE = 'ServerGroup'`.